dbms/src/Storages/S3/S3Common.cpp

```diff
@@ -126,10 +126,12 @@ void ClientFactory::init(const StorageS3Config & config_)
     config = config_;
     Aws::InitAPI(aws_options);
     Aws::Utils::Logging::InitializeAWSLogging(std::make_shared<AWSLogger>());
+    shared_client = create();
 }
 
 void ClientFactory::shutdown()
 {
+    shared_client.reset(); // Reset S3Client before Aws::ShutdownAPI.
```

Comment:
Is it a must-do that all S3Clients are released before Aws::ShutdownAPI?

As `shared_client` is a shared_ptr, I think it maybe hard for us to ensure all threads are stopped handling the S3 requests before ClientFactory::shutdown.